### PR TITLE
:bug: Delay adding the ampshare fragment parameter until after the document is first visible

### PR DESCRIPTION
**Pull request by ryjust**
When AMP CSI is enabled for CCT embedded documents, the hash fragment is updated by Chrome on click to include the click time. This process removes the ampshare parameter. The page becomes visible only after the fragment is updated so we need to delay adding the ampshare parameter until then.

### CODE REVIEW

File: test/functional/test-viewer.js [hunk 1]
```diff
@@ -180,8 +182,10 @@ describe('Viewer', () => {
     const viewer = new Viewer(ampdoc);
     expect(viewer.getParam('test')).to.equal('1');
     expect(viewer.isCctEmbedded()).to.be.true;
-    expect(windowApi.history.replaceState).to.be.calledWith({}, '',
-        '#test=1&ampshare=http%3A%2F%2Fwww.example.com%2F');
+    return Promise.resolve(() => {
```
[the remark this answers is not in the document]
Comment (ryjust) [reply test/functional/test-viewer.js]: Good suggestion! Much cleaner. Fixed!

**Comment by choumx** [comment]
Thanks for contributing!

**Comment by choumx** [comment]
Unit tests are unexpectedly failing on this test:
```
DESCRIBE => amp-a4a
  IT => should set height/width on iframe matching header value
    ✗ Error: Uncaught TypeError: win.getComputedStyle is not a function (node_modules/sinon/pkg/sinon.js:2538)
```
I'm getting the same error on #17128. The amp-a4a test probably implicitly depends on order of test execution. 😢 I'll dig into this and merge when ready.

/cc @rsimha

**Comment by rsimha** [comment]
The version of `sinon` we use is pretty old. I'm working on migrating to the latest version, which fixes some bugs, and has a more intuitive API and clearer error messages.

**Comment by choumx** [comment]
@ryjust I fixed the issue in master. Can you merge latest and push?

**Comment by ryjust** [comment]
Everything is passing now. Thanks!

**Pull request merged**